pkg/globalnet/controllers/uninstall.go

```diff
@@ -0,0 +1,178 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/submariner-io/admiral/pkg/finalizer"
+	"github.com/submariner-io/admiral/pkg/resource"
+	"github.com/submariner-io/admiral/pkg/util"
+	submarinerv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	versioned "github.com/submariner-io/submariner/pkg/client/clientset/versioned"
+	"github.com/submariner-io/submariner/pkg/globalnet/constants"
+	"github.com/submariner-io/submariner/pkg/globalnet/controllers/iptables"
+	"github.com/submariner-io/submariner/pkg/ipset"
+	routeAgent "github.com/submariner-io/submariner/pkg/routeagent_driver/constants"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes/scheme"
+	rest "k8s.io/client-go/rest"
+	"k8s.io/klog"
+	utilexec "k8s.io/utils/exec"
+)
+
+func UninstallDataPath() {
+	ipt, err := iptables.New()
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	natTableChains := []string{
+		// The chains have to be deleted in a specific order.
+		constants.SmGlobalnetEgressChainForCluster,
+		constants.SmGlobalnetEgressChainForHeadlessSvcPods,
+		constants.SmGlobalnetEgressChainForNamespace,
+		constants.SmGlobalnetEgressChainForPods,
+		constants.SmGlobalnetIngressChain,
+		constants.SmGlobalnetMarkChain,
+		constants.SmGlobalnetEgressChain,
+	}
+
+	for _, chain := range natTableChains {
+		err = ipt.FlushIPTableChain(constants.NATTable, chain)
+		if err != nil {
+			// Just log an error as this is part of uninstallation.
+			klog.Error(err)
+		}
+	}
+
+	err = ipt.FlushIPTableChain(constants.NATTable, routeAgent.SmPostRoutingChain)
+	if err != nil {
+		klog.Error(err)
+	}
+
+	if err := ipt.DeleteIPTableRule(constants.NATTable, "PREROUTING", constants.SmGlobalnetIngressChain); err != nil {
+		klog.Errorf("error deleting iptables rule for %q in PREROUTING chain: %v\n", constants.SmGlobalnetIngressChain, err)
+	}
+
+	for _, chain := range natTableChains {
+		err = ipt.DeleteIPTableChain(constants.NATTable, chain)
```

Comment:
It seems that [ClearAndDeleteChain](https://github.com/coreos/go-iptables/blob/14d56d57c892f27a717aa6026fd2d3293221395b/iptables/iptables.go#L453)  does both flush and delete. It also verifies the chain exists.